backend/passport.js

```diff
@@ -0,0 +1,76 @@
+"use strict";
+
+const passport = require("passport");
+const TwitterTokenStrategy = require("passport-twitter-token");
+const FacebookTokenStrategy = require("passport-facebook-token");
+const GoogleTokenStrategy = require("passport-google-token").Strategy;
+const config = require("./config");
+
+module.exports = function () {
+  passport.use(
+    new TwitterTokenStrategy(
+      {
+        consumerKey: config.twitterAuth.consumerKey,
+        consumerSecret: config.twitterAuth.consumerSecret,
+        includeEmail: true,
+      },
+      function (token, tokenSecret, profile, done) {
+        // console.log(profile);
+        var newUser = {
+          email: profile.emails[0].value,
+          twitterProvider: {
+            id: profile.id,
+            token: token,
+            tokenSecret: tokenSecret,
+          },
+          picture: profile.photos[0].value,
+        };
+        return done(null, newUser);
```

Comment:
Wouldn't you store data for user in database or you'll get everything at real time and it will vanish as soon as user logs out?

Reply:
Yeah, everytime we query google or facebook, we get the user's name, picture and email in real time. So changes in google and facebook will reflect in our app too. So storing user info was not needed. The user's email id is to be given to other components so that they can use it as their table's id when storing user cart, and vendor services.

Reply:
I am storing the user info in the local storage to persist reload and route changes